tacs/pytacs.py

```diff
@@ -1157,10 +1157,65 @@ def elemCallBack(
                 if k2 is None or k2 > 1e3:
                     k2 = 1e3
 
+                if propInfo.type == "PBEAM":
+                    # For PBAR, we have all we need, but a PBEAM can have
+                    # varying properties along its length, TACS doesn't support
+                    # this so we'll compute average properties and use those
+                    numPropStations = len(area)
+                    if numPropStations == 1:
+                        area = area[0]
+                        I1 = I1[0]
+                        I2 = I2[0]
+                        I12 = I12[0]
+                        J = J[0]
+                    else:
+                        xStations = propInfo.xxb
+                        area = np.trapz(area, xStations)
+                        I1 = np.trapz(I1, xStations)
+                        I2 = np.trapz(I2, xStations)
+                        I12 = np.trapz(I12, xStations)
+                        J = np.trapz(J, xStations)
+
                 con = tacs.constitutive.BasicBeamConstitutive(
                     mat, A=area, Iy=I2, Iz=I1, Iyz=I12, J=J, ky=k1, kz=k2
                 )
 
+            elif propInfo.type == "PBEAML":
+                sectionType = propInfo.beam_type
+                sectionProps = {}
+                if sectionType == "BAR":
+                    sectionProps["w"] = propInfo.dim[:, 0]
+                    sectionProps["t"] = propInfo.dim[:, 1]
+                    conType = tacs.constitutive.IsoRectangleBeamConstitutive
+                elif propInfo.Type == "TUBE":
+                    r1 = propInfo.dim[:, 0]
+                    r0 = propInfo.dim[:, 1]
+                    sectionProps["d_inner"] = 2 * r0
+                    sectionProps["t_wall"] = r1 - r0
```

Comment:
It looks like these keys are used later as arguments for the constitutive class. If that's the case, these should be "d" and "t" for IsoTubeBeamConstitutive